lib/matplotlib/widgets.py

```diff
@@ -2595,3 +2598,190 @@ def onmove(self, event):
+        *useblit* : bool
+           - Whether to use blitting.
+        *button* : list
+            The button numbers supported for the tool (defaults to [1, 2, 3])
+        *state_modifier_keys* : dict
+            A mapping of key names to state modifiers.
+        """
+        super(Painter, self).__init__(ax, on_select,
+            useblit=useblit, button=button, state_modifier_keys=None)
+        self.cmap = cmap
+        self._previous = None
+        self._overlay = None
+        self._overlay_plot = None
+        self._cursor_shape = [0, 0, 0]
+
+        props = dict(edgecolor='r', facecolor='0.7', alpha=1,
+                     animated=self.useblit, visible=False, zorder=2)
+        props.update(cursor_props or {})
+        self._cursor = Rectangle((0, 0), 0, 0, **props)
+        self.ax.add_patch(self._cursor)
+
+        x0, x1 = self.ax.get_xlim()
+        y0, y1 = self.ax.get_ylim()
+        if y0 < y1:
+            origin = 'lower'
+        else:
+            origin = 'upper'
+        props = dict(cmap=self.cmap, alpha=0.5, origin=origin,
+                     norm=NoNorm(), visible=False, zorder=1,
+                     extent=(x0, x1, y0, y1), aspect=self.ax.get_aspect())
+        props.update(overlay_props or {})
+
+        self._resize(None)
+        self._overlay_plot = self.ax.imshow(self._overlay, **props)
+        self.artists = [self._cursor, self._overlay_plot]
+        self.connect_event('resize_event', self._resize)
+
+        # These must be called last
+        self.label = 1
+        self.radius = radius
+        self._drawing = True
+        for artist in self.artists:
+            artist.set_visible(True)
+
+    @property
+    def overlay(self):
+        """The paint overlay image"""
+        return self._overlay
+
+    @overlay.setter
+    def overlay(self, image):
+        self._overlay = image
+        if image is None:
+            self.ax.images.remove(self._overlay_plot)
+            self.update()
+            return
+        self._overlay_plot.set_data(image)
+        self._shape = image.shape
+        x0, x1 = self.ax.get_xlim()
+        y0, y1 = self.ax.get_ylim()
+        self._overlay_plot.set_extent((x0, x1, y0, y1))
+        # Update the radii and window.
+        self.radius = self._radius
+        self.update()
+
+    @property
+    def label(self):
+        """The label index"""
+        return self._label
+
+    @label.setter
+    def label(self, value):
+        if value >= self.cmap.N:
+            raise ValueError('Maximum label value = %s' % len(self.cmap - 1))
+        self._label = value
+        self._cursor.set_edgecolor(self.cmap(value))
+
+    @property
+    def radius(self):
+        """The label radius in pixels"""
+        return self._radius
+
+    @radius.setter
+    def radius(self, r):
+        self._radius = r
+        xfm = self.ax.transData.inverted()
+        x0, y0 = xfm.transform((0, 0))
+        x1, y1 = xfm.transform((r, r))
```

Comment:
Added a note that polar plots are not supported (to keep the implementation simpler).